Resolver.cc:321:10: error: member reference base type 'const char' is not a structure or union

While compiling lftp 4.7.5 under macOS (10.12.2 using MacPorts) I get the following errors:

Resolver.cc:321:10: error: member reference base type 'const char' is not a structure or union
      if(IN6_IS_ADDR_LINKLOCAL(address) && scope==0) {
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/netinet6/in6.h:292:7: note: expanded from macro 'IN6_IS_ADDR_LINKLOCAL'
        (((a)->s6_addr[0] == 0xfe) && (((a)->s6_addr[1] & 0xc0) == 0x80))
          ~~~^ ~~~~~~~
Resolver.cc:321:10: error: member reference base type 'const char' is not a structure or union
      if(IN6_IS_ADDR_LINKLOCAL(address) && scope==0) {
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/netinet6/in6.h:292:37: note: expanded from macro 'IN6_IS_ADDR_LINKLOCAL'
        (((a)->s6_addr[0] == 0xfe) && (((a)->s6_addr[1] & 0xc0) == 0x80))
                                        ~~~^ ~~~~~~~

I've attached the full build log file:
[lftp-4.7.5-main.log.gz](https://github.com/lavv17/lftp/files/688543/lftp-4.7.5-main.log.gz)

--- a/src/Resolver.cc
+++ b/src/Resolver.cc
@@ -318,7 +318,7 @@ void Resolver::AddAddress(int family,const char *address,int len, unsigned int s
    case AF_INET6:
       if(sizeof(add.in6.sin6_addr) != len)
          return;
-      if(IN6_IS_ADDR_LINKLOCAL(address) && scope==0) {
+      if(IN6_IS_ADDR_LINKLOCAL((const struct in6_addr*)address) && scope==0) {
 	 error=_("Link-local IPv6 address should have a scope");
 	 return;
       }

